src/BMI/bmi_port.c

```diff
@@ -319,7 +319,9 @@ static int _bmi_port_interface_remove(bmi_port_mgr_t *port_mgr, int port_num) {
 
   char buf[1] = {'\x00'};
   write(port_mgr->socketpairfd[0], buf, sizeof(buf));
+  pthread_rwlock_unlock(&port_mgr->lock);
   read(port_mgr->socketpairfd[0], buf, sizeof(buf));
+  pthread_rwlock_wrlock(&port_mgr->lock);
```

Comment:
I agree there is a possible deadlock and it needs to be fixed, however I don't see any justification that the change is correct, and no explanation as to the placement of the `unlock` instruction (why not right after the `FD_CLR` call?).
For example, any possible issue if there is a concurrent call to `bmi_port_interface_add` with the same `port_num`?